### PR TITLE
fixes issues #18 #19

### PR DESCRIPTION
**Pull request by tech4GT**
Signed-off-by: tech4GT <varun.gupta1798@gmail.com>

### CODE REVIEW

**Comment by tech4GT** [comment]
@championswimmer sir please go through the new structure of submissions table and its associations and provide your review

File: db/actions/submissions.js [hunk 1]
```diff
@@ -6,11 +6,10 @@
 const models = require('../models');
 
 
-
 //TODO  Error: Can't set headers after they are sent.
 
 //add a submission
-function addSubmission(studentId, assnId, URL, done) {
+function addSubmission(studentId, BatchAssnId, URL, done) {
```
Comment (championswimmer) [inline db/actions/submissions.js]: variables start with small letter

File: db/actions/submissions.js [hunk 2]
```diff
@@ -22,46 +21,36 @@ function addSubmission(studentId, assnId, URL, done) {
...
-                        if (err) throw err;
-                    });
-
-                if (flag) break;
-            }
+            models.BatchAssignments.findOne({
+                where: {
+                    id: BatchAssnId
+                }
+            }).then(function (BatchAssignmentData) {
```
Comment (championswimmer) [inline db/actions/submissions.js]: variables start with small letter

File: db/actions/submissions.js [hunk 3]
```diff
@@ -110,18 +98,17 @@ function acceptSubmissionbyId(id, echo, done) {
 }
 
 //function to accept a submission without submission id
-function acceptSubmissionWithoutId(studentId, assnId, URL, done) {
+function acceptSubmissionWithoutId(studentId, BatchAssnId, URL, done) {
     models.Submissions
```
Comment (championswimmer) [inline db/actions/submissions.js]: variables start with small letter

File: routes/api_v1/submissions.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ router.post('/new', function(req, res) {
   if (req.body.URL) {
     db.actions.submissions.addSubmission(
       req.body.studentId,
-      req.body.assignmentId,
+      req.body.BatchAssignmentId,
```
Comment (championswimmer) [inline routes/api_v1/submissions.js]: req arguments should also be with small

File: routes/api_v1/submissions.js [hunk 5]
```diff
@@ -72,13 +72,13 @@ router.get('/', function(req, res) {
...
+    if (req.query.student && req.query.BatchAssignment) {
       options.studentId = req.query.student;
-      options.assignmentId = req.query.assignment;
+      options.batch_assignmentId = req.query.BatchAssignment;
     } else if (req.query.student) {
       options.studentId = req.query.student;
-    } else if (req.query.assignment) {
-      options.assignmentId = req.query.assignment;
+    } else if (req.query.BatchAssignment) {
+      options.batch_assignmentId = req.query.BatchAssignment;
```
Comment (championswimmer) [inline routes/api_v1/submissions.js]: either use camelCase or underscore_case. not both

**Review by championswimmer — CHANGES_REQUESTED** [review]

**Comment by tech4GT** [comment]
@championswimmer sir done all the changes

File: db/models.js [hunk 6]
```diff
@@ -60,18 +71,11 @@ const Batches = sequelize.define('batch', {
 const Submissions = sequelize.define('submission', {
     id: {type: Sequelize.INTEGER, primaryKey: true, autoIncrement: true},
     status: Sequelize.BOOLEAN,
-    URL: Sequelize.STRING
```
Comment (championswimmer) [inline db/models.js:63]: dont use ALL CAPS keys in objects.

File: routes/api_v1/submissions.js [hunk 7]
```diff
@@ -23,7 +23,7 @@ router.post('/new', function(req, res) {
   if (req.body.URL) {
     db.actions.submissions.addSubmission(
       req.body.studentId,
-      req.body.assignmentId,
+      req.body.batch_assignmentId,
```
Comment (championswimmer) [inline routes/api_v1/submissions.js]: camel and underscore mixed

File: routes/api_v1/submissions.js [hunk 8]
```diff
@@ -72,13 +72,13 @@ router.get('/', function(req, res) {
 
     if (accepted) options.status = JSON.parse(accepted);
 
-    if (req.query.student && req.query.assignment) {
+    if (req.query.student && req.query.batch_assignment) {
       options.studentId = req.query.student;
-      options.assignmentId = req.query.assignment;
+      options.batch_assignmentId = req.query.batch_assignment;
```
Comment (championswimmer) [inline routes/api_v1/submissions.js]: camel underscore mixed

**Review by championswimmer — CHANGES_REQUESTED** [review]

**Comment by tech4GT** [comment]
@championswimmer worked out these ones as well

**Review by championswimmer — APPROVED** [review]

**Pull request merged**